### PR TITLE
Replace asserts with raise exceptions

### PR DESCRIPTION
**Pull request by banasiakadam60**

### CODE REVIEW

File: golem/docker/task_thread.py [hunk 1]
```diff
@@ -26,11 +26,12 @@ def __init__(self, task_computer, subtask_id, docker_images,
                  orig_script_dir, src_code, extra_data, short_desc,
                  res_path, tmp_path, timeout, check_mem=False):
 
+        if docker_images is None:
```
Comment (mfranciszkiewicz) [inline golem/docker/task_thread.py]: `if not docker_images` would handle empty lists

File: golem/network/transport/message.py [hunk 2]
```diff
@@ -351,7 +354,8 @@ def __init__(self, sig="", timestamp=None, dict_repr=None):
         Message.__init__(self, MessagePing.Type, sig, timestamp)
 
         if dict_repr:
-            assert dict_repr.get(MessagePing.PING_STR)
+            if dict_repr.get(MessagePing.PING_STR) is None:
```
Comment (mfranciszkiewicz) [inline golem/network/transport/message.py]: You can use `if not ...` to catch empty values

File: golem/task/taskmanager.py [hunk 3]
```diff
@@ -90,9 +91,12 @@ def unregister_listener(self, listener):
             logger.warning("Trying to unregister listener that wasn't registered {}".format(listener))
 
     def add_new_task(self, task):
-        assert task.header.task_id not in self.tasks
-        assert self.key_id
-        assert SocketAddress.is_proper_address(self.listen_address, self.listen_port)
+        if task.header.task_id in self.tasks:
+            raise RuntimeError("Task has been already added")
+        if self.key_id is None:
```
Comment (mfranciszkiewicz) [inline golem/task/taskmanager.py]: `if not self.key_id:`

File: tests/golem/interface/test_cli.py [hunk 4]
```diff
@@ -163,7 +163,7 @@ def command(self):
 
             with patch('sys.stderr', new_callable=StringIO) as err:
                 cli.process(['commands', 'command'])
-                assert not err.getvalue()
+                self.assertIsNotNone(err.getvalue())
```
Comment (mfranciszkiewicz) [inline tests/golem/interface/test_cli.py]: This test expects that no messages were written to `sys.stderr`. `assertIsNotNone` does not check whether it is an empty string. The same goes for lines below.

File: tests/golem/interface/test_cli.py [hunk 5]
```diff
@@ -301,23 +301,22 @@ class TestCLICommands(unittest.TestCase):
...
+        self.assertIsNotNone(CommandHelper.get_interface(_help))
 
         with self.assertRaises(ParsingException):
             _help()
 
     @patch('sys.stdout', new_callable=MockStdout)
     def test_debug(self, out):
         _debug()
-        assert out.getvalue()
+        self.assertIsNotNone(out.getvalue())
```
Comment (mfranciszkiewicz) [inline tests/golem/interface/test_cli.py]: Does not check for empty values.

File: tests/golem/interface/test_formatters.py [hunk 6]
```diff
@@ -52,21 +53,19 @@ def test_namespace(self):
         fmt = CommandFormatter()
         fmt_json = CommandJSONFormatter()
 
-        assert not fmt.supports(ns_dict)
-        assert not fmt_json.supports(ns_dict)
+        self.assertIsNone(fmt.supports(ns_dict))
```
Comment (mfranciszkiewicz) [inline tests/golem/interface/test_formatters.py]: `supports` returns a `bool`ean

**Review by mfranciszkiewicz — CHANGES_REQUESTED** [review]

File: golem/docker/task_thread.py (hunk 1 above)
Comment (banasiakadam60) [reply golem/docker/task_thread.py]: Done

File: golem/network/transport/message.py (hunk 2 above)
Comment (banasiakadam60) [reply golem/network/transport/message.py]: Done

File: golem/task/taskmanager.py (hunk 3 above)
Comment (banasiakadam60) [reply golem/task/taskmanager.py]: Done

File: tests/golem/interface/test_cli.py (hunk 4 above)
Comment (banasiakadam60) [reply tests/golem/interface/test_cli.py]: Fixed

File: tests/golem/interface/test_cli.py (hunk 5 above)
Comment (banasiakadam60) [reply tests/golem/interface/test_cli.py]: Done

File: tests/golem/interface/test_formatters.py (hunk 6 above)
Comment (banasiakadam60) [reply tests/golem/interface/test_formatters.py]: Fixed

**Pull request merged**